### DISCUSSION TITLE
Build failing on Mac Mini M1

### DISCUSSION THREAD

**Discussion by Alvaro948** [codemagic-ci-cd/codemagic-docs#1163] [Q&A] [answered]
Seems that codemagic no longer accepts `mac_mini` instance. Copy & paste from https://github.com/orgs/codemagic-ci-cd/discussions/933#discussioncomment-4045477

After setting instance type from `mac_mini` to `mac_mini_m1` the build is failing on ionic capacitor:

```yaml
      - name: Build ionic app
        script: |
          ionic cap build android --configuration production
```

![image](https://user-images.githubusercontent.com/97282850/199691385-69d9c5df-1eab-48fb-8e7e-60fd5e5b8603.png)

Failed build: 63638b8db83454e5368c79ca
Today's build (failing) 6374adde95f1338a7e915a06
Successful build: 63638c6a49418626798bb6de (after reverting the instance type back to `mac_mini`)

![image](https://user-images.githubusercontent.com/97282850/199691165-647fb4fe-5df1-4c42-a96f-2014cf2a63d0.png)

**Comment by himesh-cm** [comment]
@Alvaro948, Thanks for sharing the build info, I will check and get back to you.

**Comment by himesh-cm** [comment, marked as answer]
Hi @Alvaro948,
The error from the logs are, which fails at step - `Build Ionic App`
```
[ERROR] Unable to find command: cap build android production

  ionic cap - Capacitor functionality

    These commands integrate with Capacitor, Ionic's new native layer project which provides an alternative to Cordova for native functionality in your app.

Project Commands:

    add ............................. Add a native platform to your Ionic project
    copy ............................ Copy web assets to native platforms
    open ............................ Open the IDE for a given native platform project
    run ............................. (beta) Run an Ionic project on a connected device
    sync ............................ Sync (copy + update) an Ionic project
    update .......................... Update Capacitor native platforms, install Capacitor/Cordova plugins
 ```

Can you please check this once?

**Reply by Alvaro948** [reply]
Hi @himesh-cm , yes, that is exactly what I was refering to. For some reason the instance `mac_mini_m1` fails on executing that command, but the `mac_mini` instance did not (same as my local environment).

Failed build: 63638b8db83454e5368c79ca
Today's build (failing) 6374adde95f1338a7e915a06
Successful build: 63638c6a49418626798bb6de (after reverting the instance type back to mac_mini)

Note that executions mentioned are exactly the same code & config, just chaging the instance type.

**Reply by himesh-cm** [reply]
@Alvaro948 
I think its related to Ionic version - checking it [here](https://docs.codemagic.io/specs/versions-macos/)
Mac mini - has Ionic version: `ionic 6.12.4` while M1 has `ionic 5.4.16`

I will take this up with the team, but meanwhile, can you upgrade the Ionic version at the top of your workflow and try the build again?
Apologies for the inconvenience caused. 

Please let us know how it goes.

**Reply by himesh-cm** [reply]
Hi @Alvaro948, Can you please confirm, if you were able to try the above fix?

**Reply by Alvaro948** [reply]
Hi @himesh-cm ,

Sorry for the delay. Yes!, it worked by using the npminstalled version of `@ionic/cli`:

```yaml
      - name: Build ionic app
        script: |
          npx ionic cap build android --configuration production
```

Thank you so much for your assistence, and the platform, you are awesome!
